src/main/java/rip/bolt/ingame/ranked/PlayerWatcher.java

```diff
@@ -140,7 +133,7 @@ public void onMatchStart(MatchStartEvent event) {
 
     // If a player never joined mark as abandoned
     if (playersAbandoned(getNonJoinedPlayers())) {
-      cancelMatch(event.getMatch());
+      rankedManager.cancel(event.getMatch(), CancelReason.AUTOMATED_CANCEL);
       event.getMatch().sendMessage(Messages.matchStartCancelled());
     }
```

Comment:
Here could iterate players to find who left, then call for `matchParticipation.playerLeft()` for each one, and then call startCountdownIfRequired with the list of uuids as 2nd parameter.

Then make it so startCountdownIfRequired can take a list of uuids, if passed, it uses that list directly, if not passed, it does the logic of finding the player who left for longest.
Then make LeaveCountdown switch from UUID to List<UUID>, you already have playersAbandoned which you can call with that list of uuids to ban everyone so it shouldn't be a big change.